packages/serverpod_client/lib/src/serverpod_client_io.dart

```diff
@@ -70,11 +69,27 @@ abstract class ServerpodClient extends ServerpodClientShared {
       var data = await _readResponse(response);
 
       if (response.statusCode != HttpStatus.ok) {
-        throw getExceptionFrom(
-          data: data,
-          serializationManager: serializationManager,
-          statusCode: response.statusCode,
-        );
+        if (data.isEmpty) {
+          throw switch (response.statusCode) {
+            HttpStatus.badRequest => throw ServerpodClientException(
+                'Bad request', response.statusCode),
+            HttpStatus.unauthorized => throw ServerpodClientException(
+                'Unauthorized', response.statusCode),
+            HttpStatus.forbidden =>
+              throw ServerpodClientException('Forbidden', response.statusCode),
+            HttpStatus.notFound =>
+              throw ServerpodClientException('Not found', response.statusCode),
+            HttpStatus.internalServerError => throw ServerpodClientException(
+                'Internal server error', response.statusCode),
+            _ => ServerpodClientException('Unknown error', response.statusCode),
```

Comment:
Suggestion: Move this code into the method `getExceptionFrom` as it needs to work in web as well and that method is shared between io and the web implementation. 

Then I would also like to see Explicit exception types for each exception here, e.g.

```dart
class ServerpodClientBadRequest extends ServerpodClientException {
  ServerpodClientBadRequest(int statusCode) : super('Bad request', statusCode);
}
```